integration/test_session_error.py

```diff
@@ -0,0 +1,178 @@
+        return olm_message.ciphertext
+
+    async def _handle_message_get_one_time_key_resp(
+        self, message: CansMessage
+    ) -> None:
+        """Handle message type GET_ONE_TIME_KEY_RESP."""
+        await super()._handle_message_get_one_time_key_resp(message)
+        # Successfully reset the session after session error
+        raise SessionErrorOkException()
+
+
+class MockClient(Client):
+    """Mock client."""
+
+    def __init__(
+        self,
+        name: str,
+        keys: EcPemKeyPair,
+        friends: Set,
+        crypto_session_peer_override: Optional[str] = None,
+    ) -> None:
+        """Construct mock client."""
+        self.server_hostname = os.environ["CANS_SERVER_HOSTNAME"]
+        self.server_port = os.environ["CANS_PORT"]
+        self.certpath = os.environ["CANS_SELF_SIGNED_CERT_PATH"]
+        self.event_loop = asyncio.get_event_loop()
+        self._do_logger_config()
+        self.name = name
+
+        self.log = logging.getLogger("cans-logger")
+        self.log.setLevel(logging.DEBUG)
+
+        self.friends = friends
+
+        account = Account()
+        self.session_manager = MockSessionManager(
+            keys=keys,
+            account=account,
+            crypto_session_peer_override=crypto_session_peer_override,
+        )
+
+    async def run(self) -> None:
+        """Run dummy client application."""
+        # Connect to the server
+        await self.session_manager.connect(
+            url=f"wss://{self.server_hostname}:{self.server_port}",
+            certpath=self.certpath,
+            friends=self.friends,
+        )
+
+    async def ping_peers(self) -> None:
+        """Continually ping your peers."""
+        while True:
+            await asyncio.sleep(1)
+            for peer in self.friends:
+                message, _ = self.session_manager.user_message_to(
+                    peer, f"Hello {peer}"
+                )
+                await self.session_manager.send_message(message)
+
+
+async def impl_test_bad_message_format():
+    """Async implementation of the test."""
+    alice_secret, alice_public = generate_keys()
+    bob_secret, bob_public = generate_keys()
+    mallory_secret, mallory_public = generate_keys()
+
+    # Instantiate three parties and let them all be friends with each other
+    alice = MockClient(
+        name="Alice",
+        keys=(alice_secret, alice_public),
+        friends={digest_key(bob_public), digest_key(mallory_public)},
+        crypto_session_peer_override=None,
+    )
+    bob = MockClient(
+        name="Bob",
+        keys=(bob_secret, bob_public),
+        friends={digest_key(alice_public), digest_key(mallory_public)},
+        crypto_session_peer_override=None,
+    )
+    mallory = MockClient(
+        name="Charlie",
+        keys=(mallory_secret, mallory_public),
+        friends={digest_key(alice_public), digest_key(bob_public)},
+        # Let Mallory be the broken client that always uses his session
```

Comment:
this one https://pl.wikipedia.org/wiki/George_Mallory ? :)
I guess in the high mountains one may get confused and change his name from George to Mallory

Reply:
<3